index.js

```diff
@@ -195,26 +195,13 @@ NYC.prototype.instrumentAllFiles = function (input, output, cb) {
   try {
     const stats = fs.lstatSync(input)
     if (stats.isDirectory()) {
-      const globOptions = { dot: true, mark: true, ignore: ['**/.git', '**/.git/**/*'] }
-      const outputPaths = (output) ? glob.sync(`${path.resolve(input)}/**/*`, globOptions) : []
-
-      const partition = (universal, subsetFilter) => {
-        return universal.reduce(([a, aDash], member) => {
-          return subsetFilter(member) ? [[...a, member], aDash] : [a, [...aDash, member]]
-        }, [[], []])
-      }
-
-      const [dirs, files] = partition(outputPaths, filename => filename.endsWith('/'))
-
       inputDir = input
       this.walkAllFiles(input, visitor)
 
-      if (files.length || dirs.length) {
-        dirs.map(file => path.resolve(output, path.relative(input, file)))
-          .forEach(dir => mkdirp.sync(dir))
-
-        files.map(file => path.relative(input, file))
-          .forEach(file => { copySync(path.join(input, file), path.join(output, file), { overwrite: false }) })
+      if (output) {
+        const globOptions = { dot: true, ignore: ['**/.git', `**/${path.basename(output)}`] }
```

Comment:
I suspect this is a problem for `nyc instrument . output` if `./lib/output/data.json` exists (that output folder should be copied).  Instead of ignoring `**/${path.basename(output)}` couldn't we just ignore `output`?

Reply:
This only produces a list of top level files using ``glob.sync(`${path.resolve(input)}/*`, ...)``, so the inclusion of ``**/${path.basename(output)}`` won't exclude any further sub-dirs.  The recursive copy of each top level sub-dir is then handled by `fs-extra.copySync`.  I agree it looks a little silly and should really be just `output`, but I couldn't get it to work with `glob.sync`.  I should point out that this also means we're only ignoring `.git` directories at the top level in this implementation.

Anyway I'll see if I can come up with a nicer solution here, in the meantime I'll drop in the change below and rebase

Reply:
What if someone does `nyc instrument . subdir/output`?  In that case wouldn't we need to ignore `subdir`, not `output`?  I ask due to the use of `path.basename`.